### DISCUSSION TITLE
windows VM boot

### DISCUSSION THREAD

**Discussion by Mingwang-Li** [tianocore/edk2#4978] [Q&A]
A blue screen of death (BSOD) of watch dog timeout occurs during Windows VM boot. The Windows dump log shows that some stacks are in efi. What is the cause?

windows stack：
KiFreezeTargetExecution+0x10c
KiIpiServiceRoutine+0x22
KiPlayInterrupt+0x208
KiProcessInterrupt+0x178
KiKernelSp0InterruptHandler+0x5c
HalEfiSetEnvironmentVariable+0xcc
HalSetEnvironmentVariableEx+0x16c
IoSetEnvironmentVariableEx+0x7c
ExpSetBootEntry+0x6b0
KiSystemServiceCopyEnd+0x38
KiServiceInternal+0x60
BiAddBootEntry+0x40
BiCreateEfiEntry+0x98
BiBindEfiEntries+0xd4
BiBindEfiNamespaceObjects+0x50
BiBindFirmwareToSystemStore+0x40
BiOpenSystemStore+0x1c8
BcdOpenStore+0xc4
PoInitHiberServices+0xc0
CmCompleteRegistryInitialization+0x170
NtInitializeRegistry+0xac
KiSystemServiceCopyEnd+0x38
KiServiceInternal+0x60
NtInitializeRegistry+0x44
KiSystemServiceCopyEnd+0x38
KiSystemServiceExit

**Comment by ajfish** [comment]
EFI has runtime services that allows an OS to write to NVRAM. These runtime services get called in a virtual address space mapping built by the kernel.

Thanks,

Andrew Fish
